lotus/settings.py

```diff
@@ -15,11 +17,21 @@ class Settings:
     # Serialization setting
     serialization_format: SerializationFormat = SerializationFormat.DEFAULT
 
+    # Multithreading settings
+    enable_multithreading: bool = False
```

Comment:
Not super important, but just pointing out that `litellm` uses multi-threading under the hood to send parallel requests to APIs. So, even if someone specified `enable_multithreading=False`, there will still be multiple threads spawned by the process.

Reply:
Should we have this as a setting at all? Or just always enable multithreading for groups

Reply:
I'm not sure, maybe always enable it? Unless you see some major downsides. I suppose one issue is rate limiting

Reply:
One idea might be to add an option for the maximum number of threads that can be spawned -- if we have too high of a cardinality in the group-by couldn't this exhaust the system resources?

Reply:
Yeah I was thinking the same - maybe we can set a general default to 8 threads.

Reply:
Resolved